libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushService.kt

```diff
@@ -413,8 +414,32 @@ open class PushService {
     companion object {
         private const val TAG = "PushService"
 
-        // Retry time constants.
-        private const val MILLISECONDS_IN_SIX_DAYS = 518400000L
+        /**
+         * Specifies if push notification (re-)registration should occur one
+         * time when the app is brought to the foreground.
+         *
+         * This is mutually exclusive with
+         * [isPushNotificationsRegisterPeriodicallyEnabled] and is the
+         * default option of the two.
+         */
+        var isPushNotificationsRegistrationOneTimeOnAppForegroundEnabled = true
```

Comment:
Same comment as above. Instead of two separate flags (which gives us 4 possible states), could we have an enum with 3 values: no-re-registration, re-registration on foreground, periodic re-registration.
Again we could offer three methods to pick the "mode".